src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetBrokeredServices.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public static class NuGetBrokeredServices
+    {
+        public const string SourceRepositoryProviderServiceName = "NuGet.SourceRepositoryProviderService";
+        public const string SourceRepositoryProviderServiceVersion = "1.0.0";
+
+
+        public static readonly ServiceRpcDescriptor SourceRepositoryProviderService = new ServiceJsonRpcDescriptor(
+              new ServiceMoniker(SourceRepositoryProviderServiceName, new Version(SourceRepositoryProviderServiceVersion)),
+              ServiceJsonRpcDescriptor.Formatters.UTF8,
+              ServiceJsonRpcDescriptor.MessageDelimiters.HttpLikeHeaders);
+
+        public static BrokeredServiceFactory GetSourceRepositoryProviderServiceFactory()
+        {
+            return (mk, options, sb, ct) => new ValueTask<object>(new NuGetSourceRepositoryService());
```

Comment:
It may be a good idea to pass in `options, sb` and the authorization service to the `NuGetSourceRepositoryService` constructor, even if you don't need it yet, just to set the next engineer up for when they change `NuGetSourceRepositoryService` in a way that requires any of these things.
For example we would *not* want that new change to acquire the global brokered service container -- they should absolutely use the `sb` given here.